lib/acyclicGraph.ml

```diff
@@ -608,21 +659,27 @@ module Make (Point:Point) = struct
 
   let check_eq g u v =
     u == v ||
-    let arcu = repr g u and arcv = repr g v in
+    let arcu = repr_node g u and arcv = repr_node g v in
     arcu == arcv
 
   let check_smaller g strict u v =
-    search_path strict (repr g u) (repr g v) g
+    search_path strict (repr_node g u) (repr_node g v) g
 
   let check_leq g u v = check_smaller g false u v
   let check_lt g u v = check_smaller g true u v
 
   (* enforce_eq g u v will force u=v if possible, will fail otherwise *)
 
-  let rec enforce_eq u v g =
-    let ucan = repr g u in
-    let vcan = repr g v in
-    if topo_compare ucan vcan = 1 then enforce_eq v u g
+  let enforce_eq u v g =
+    let ucan = repr_node g u in
+    let vcan = repr_node g v in
+    if ucan == vcan then g
+    else if topo_compare ucan vcan = 1 then
+      let ucan = vcan and vcan = ucan in
```

Comment:
Why not do `let ucan, vcan = if topo_compare ucan vcan = 1 then vcan, ucan else ucan, vcan in` instead of duplicating the branches here?

Reply:
Probably better indeed.

Reply:
I remember why I did that: the error case also needs u and v swapped in the application of `Point.error_inconsistency`. It's a bit cumbersome to recompute the order, while the code duplication here is relatively small and makes clear the intention.

Reply:
>the error case also needs u and v swapped

Does it really?

Reply:
ok